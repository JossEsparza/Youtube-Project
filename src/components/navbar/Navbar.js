import React, { useState } from 'react';
import { useContext } from 'react';
import { FaSearch, FaRegUserCircle, FaHamburger, FaAngleLeft } from 'react-icons/fa';
import { ToggleContext } from '../../state/ToggleContext';
import { useHistory } from 'react-router-dom';
import './navbar.scss';
import './dark_navbar.scss';
import './light_navbar.scss';

const Navbar = () => {
  const [search, setSearch] = useState('');
  const { globalReducer, dispatch } = useContext(ToggleContext);

  const handleInputChange = ({ target }) => {
    setSearch(target.value);
  };

  const navigate = useHistory();

  const handleSubmit = (e) => {
    e.preventDefault();
    navigate.push('/');
    const action = {
      type: 'set-input',
      payload: search,
    };

    dispatch(action);
    setSearch('');
  };

  const handleToggle = () => {
    const action = {
      type: 'toggle',
      payload: !globalReducer.currentToggle,
    };
    dispatch(action);
  };

  const handleThemeToggle = () => {
    const action = {
      type: 'toggle-theme',
      payload: !globalReducer.isEnabled,
    };
    dispatch(action);
  };

  return (
    <div className="navbar_container">
      <button onClick={handleToggle} className="menu_button icon">
        {globalReducer.currentToggle ? <FaAngleLeft /> : <FaHamburger />}
      </button>
      <div className="search_container">
        <form onSubmit={handleSubmit}>
          <input value={search} onChange={handleInputChange} placeholder="Search" />
          <FaSearch className="search_icon" />
        </form>
      </div>
      <div className="mode_profile_container icon">
        <label className="switch">
          <input type="checkbox" onChange={handleThemeToggle} />
          <div className="slider" />
        </label>
        <p>Dark mode</p>
        <FaRegUserCircle />
      </div>
    </div>
  );
};

export default Navbar;
