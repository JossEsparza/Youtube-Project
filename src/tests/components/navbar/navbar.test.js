import { mount } from 'enzyme';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Navbar from '../../../components/navbar/Navbar';
import { ToggleContext } from '../../../state/ToggleContext';

describe('Tests the Navbar', () => {
  test('should be displayed correctly', () => {
    const globalReducer = {
      inputValue: 'changuitos',
      currentToggle: false,
      isEnabled: false,
    };
    const wrapper = mount(
      <ToggleContext.Provider
        value={{
          globalReducer,
        }}
      >
        <Router history={createMemoryHistory()}>
          <Navbar />
        </Router>
      </ToggleContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
