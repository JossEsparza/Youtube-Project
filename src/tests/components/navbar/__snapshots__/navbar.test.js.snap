// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Navbar should be displayed correctly 1`] = `ReactWrapper {}`;
